addon/services/observer-manager.js

```diff
@@ -23,4 +23,9 @@ export default class ObserverManagerService extends Service {
   willDestroy() {
     this._admin.destroy();
   }
+
+  removeElement(element) {
+    this._admin.elementRegistry.removeElement(element);
+    this._admin.registry.removeElement(element);
```

Comment:
Where is `registry` defined on the intersection-observer-admin class?

Reply:
Yea, it seems like maybe there's some extra steps missing from the `unobserve` implementation in intersection-observer-admin to release these references.